src/json-schema.ts

```diff
@@ -0,0 +1,673 @@
+/**
+ * JSON Schema Integration for Interface-Forge
+ *
+ * This module provides functionality to automatically create Factory instances from JSON Schema definitions.
+ * It requires optional peer dependencies 'ajv' and 'ajv-formats' to be installed.
+ *
+ * @example
+ * ```typescript
+ * import { createFactoryFromJsonSchema } from 'interface-forge/json-schema';
+ *
+ * const userSchema = {
+ *   type: 'object',
+ *   properties: {
+ *     id: { type: 'string', format: 'uuid' },
+ *     name: { type: 'string', minLength: 1 },
+ *     email: { type: 'string', format: 'email' }
+ *   },
+ *   required: ['id', 'name', 'email']
+ * };
+ *
+ * const UserFactory = await createFactoryFromJsonSchema(userSchema);
+ * const user = UserFactory.build();
+ * const users = UserFactory.batch(10);
+ * ```
+ */
+
+import type { Faker } from '@faker-js/faker';
+import type { ValidateFunction } from 'ajv';
+import { isArray, isDefined, isRecord } from '@tool-belt/type-predicates';
+import { Factory } from './index.js';
+
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call, @typescript-eslint/restrict-plus-operands, @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/no-unnecessary-condition */
+
+/**
+ * Configuration options for JSON Schema factory creation
+ */
+export interface JsonSchemaOptions {
```

Comment:
extend the base factory options. Change the base factory opts into an interface and extend them